src/grad.jl

```diff
@@ -65,6 +65,17 @@ function jacobian(fdm, f, x::Union{T, AbstractArray{T}}; len::Int=length(f(x)))
     return (J, )
 end
 
+# if not number or array
+function jacobian(fdm, f, x; len::Int=length(f(x)))
+    vec_x, vec_to_x = to_vec(x)
+    return jacobian(fdm, x->f(vec_to_x(vec_x)), vec_x; len=len)
```

Comment:
Should this be `x->f(vec_to_x(x))` here?

Reply:
oh yeah, nice catch!